core/kogitoservice/deployer_resources.go

```diff
@@ -217,3 +216,49 @@ func (s *serviceDeployer) mountEnvsOnDeployment(deployment *appsv1.Deployment) {
 	deployment.Spec.Template.Spec.Containers[0].Env = framework.EnvOverride(deployment.Spec.Template.Spec.Containers[0].Env, framework.CreateEnvVar(infrastructure.RuntimeTypeKey, string(s.instance.GetSpec().GetRuntime())))
 	deployment.Spec.Template.Spec.Containers[0].Env = framework.EnvOverride(deployment.Spec.Template.Spec.Containers[0].Env, s.definition.Envs...)
 }
+
+// CreateServiceComparator creates a new comparator for Service only checking required fields
+func (s *serviceDeployer) CreateServiceComparator() func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	return func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+		svcDeployed := deployed.(*corev1.Service).DeepCopy()
+		svcRequested := requested.(*corev1.Service)
+
+		// Remove generated fields from deployed version, when not specified in requested object
+		for _, portRequested := range svcRequested.Spec.Ports {
+			if found, portDeployed := findServicePort(portRequested, svcDeployed.Spec.Ports); found {
+				if portRequested.Protocol == "" {
+					portDeployed.Protocol = ""
+				}
+			}
+		}
+		// Ignore empty label maps
+		if svcRequested.GetLabels() == nil && svcDeployed.GetLabels() != nil && len(svcDeployed.GetLabels()) == 0 {
+			svcDeployed.SetLabels(nil)
+		}
+
+		var pairs [][2]interface{}
+		pairs = append(pairs, [2]interface{}{svcDeployed.Name, svcRequested.Name})
+		pairs = append(pairs, [2]interface{}{svcDeployed.Namespace, svcRequested.Namespace})
+		pairs = append(pairs, [2]interface{}{svcDeployed.Labels, svcRequested.Labels})
+		pairs = append(pairs, [2]interface{}{svcDeployed.Spec.Ports, svcRequested.Spec.Ports})
+		pairs = append(pairs, [2]interface{}{svcDeployed.Spec.Selector, svcRequested.Spec.Selector})
+		pairs = append(pairs, [2]interface{}{svcDeployed.Spec.Type, svcRequested.Spec.Type})
+		equal := compare.EqualPairs(pairs)
+
+		// TODO: Figure out how to log differences
```

Comment:
If you do, we will have to add it to every log message we have in this code base. :sweat_smile: 
Don't worry, if you prefer, we can open a JIRA to track this down in future sprints.

Reply:
You are just referring to the TODO comment right? I just forgot to delete that line, since prior to this commit I wasn't able to get logging to work 😅 ... I just deleted the comment.